src/lightning/pytorch/core/module.py

```diff
@@ -710,11 +725,13 @@ def validation_step(self, *args: Any, **kwargs: Any) -> Optional[STEP_OUTPUT]:
             batch: The output of your :class:`~torch.utils.data.DataLoader`.
             batch_idx: The index of this batch.
             dataloader_idx: The index of the dataloader that produced this batch.
-                (only if multiple val dataloaders used)
+                (only if multiple dataloaders used)
 
         Return:
-            - Any object or value
-            - ``None`` - Validation will skip to the next batch
+            - :class:`~torch.Tensor` - The loss tensor
+            - ``dict`` - A dictionary. Can include any keys, but must include the key ``'loss'``.
```

Comment:
Maybe it doesn't make sense for val/test/predict to possibly return a loss tensor?

Reply:
Yes, there is no longer any special functionality for val/test-steps returning an object.

Reply:
In that case should we change the output type to None?

Reply:
Yes that or maybe `Any | None`. In theory the user can use the LightningModule also standalone (e.g. with Fabric) and returning something there could maybe make sense. @carmocca confirm? :)

Reply:
I believe `Any | None` is equivalent to `Any`